app/modules/events/event.ts

```diff
@@ -0,0 +1,19 @@
+import type { Event, Group, User } from '@prisma/client';
+
+interface FullEvent extends Event {
+  group?: Group;
+  users?: User[];
+}
+
+export function eventDataPatcher(event: any): FullEvent {
```

Comment:
Instead of using `any` type, one idea is to define a `RawEvent` type that has all the properties of an `Event` except for the `date` type being a `string` instead of a `Date` object.

We can achieve this by using a mapped type. Here's an example:

```ts
type RawEvent = {
  [key in keyof Event]: key extends 'date' ? string : Event[key];
}; 
```

Reference: https://www.typescriptlang.org/docs/handbook/2/mapped-types.html

Reply:
`type RawEvent = Omit<Event, 'date'> & { date: string };`

used ^ ; co-pilot recommandation

Reply:
Oh, that's even better! :)